tests/jenkins/TestDockerReRelease.groovy

```diff
@@ -0,0 +1,93 @@
+import static com.lesfurets.jenkins.unit.global.lib.GitSource.gitSource
+import static com.lesfurets.jenkins.unit.global.lib.GitSource.gitSource
+import static com.lesfurets.jenkins.unit.MethodCall.callArgsToString
+
+
+class TestDockerReRelease extends BuildPipelineTest {
+
+    @Override
+    @Before
+    void setUp() {
+
+        helper.registerSharedLibrary(
+            library().name('jenkins')
+                .defaultVersion('5.7.1')
+                .allowOverride(true)
+                .implicit(true)
+                .targetPath('vars')
+                .retriever(gitSource('https://github.com/opensearch-project/opensearch-build-libraries.git'))
+                .build()
+            )
+        super.setUp()
+
+        // Variables
+        binding.setVariable('PRODUCT', 'opensearch')
+        binding.setVariable('TAG', '1')
+        binding.setVariable('RE_RELEASE', 'true')
+
+        def inputManifest = "tests/jenkins/data/opensearch-1.3.0.yml"
+        helper.registerAllowedMethod('readYaml', [Map.class], { args ->
+            return new Yaml().load((inputManifest as File).text)
+        })
+        helper.addShMock("""docker inspect --format '{{ index .Config.Labels "org.label-schema.version"}}' opensearchproject/opensearch:1""") { script ->
+            return [stdout: "1.3.0", exitValue: 0]
+        }
+        helper.addShMock("""docker inspect --format '{{ index .Config.Labels "org.label-schema.description"}}' opensearchproject/opensearch:1""") { script ->
+            return [stdout: "7756", exitValue: 0]
+        }
+        helper.addShMock("""docker inspect --format '{{ index .Config.Labels "org.label-schema.build-date"}}' opensearchproject/opensearch:1""") { script ->
+            return [stdout: "2023-06-19T19:12:59Z", exitValue: 0]
+        }
+        helper.addShMock("""docker inspect --format '{{ index .Config.Labels "org.label-schema.version"}}' opensearchproject/opensearch:latest""") { script ->
+            return [stdout: "2.5.0", exitValue: 0]
+        }
+
+    }
+
+    @Test
+    void testReRelease() {
+        super.testPipeline('jenkins/docker/docker-re-release.jenkinsfile',
+                'tests/jenkins/jenkinsjob-regression-files/docker/docker-re-release.jenkinsfile')
+    }
+
+    @Test
+    void checkForTriggeredJobs(){
+        runScript('jenkins/docker/docker-re-release.jenkinsfile')
+        assertThat(getCommandExecutions('build', ''), hasItem('{job=docker-build, propagate=true, wait=true, parameters=[null, null, null]}'))
+        assertThat(getCommandExecutions('build', ''), hasItem('{job=docker-copy, propagate=true, wait=true, parameters=[null, null, null, null]}'))
+        assertThat(getCommandExecutions('build', ''), hasItem('{job=docker-scan, propagate=true, wait=true, parameters=[null]}'))
+        assertThat(getCommandExecutions('build', ''), hasItem('{job=docker-promotion, propagate=true, wait=true, parameters=[null, null, null]}'))
```

Comment:
Why are the parameters going `null` here? Shouldn't they be actual parameters?

Reply:
Not sure why the parameters are taken as null though the above lines shows the exact parameter values that are passed. This is the same case when the jenkins job is triggered inside the library checkout previous implementation [here](https://github.com/opensearch-project/opensearch-build-libraries/blob/560f671d486af6e78af1bb21bc7591c5388184d1/tests/jenkins/jobs/BuildDockerImage_Jenkinsfile_builds_both.txt#L15)

Reply:
Can you try `addParam` instead of binding variable? Binding variables are used for mocking env variables mostly. https://github.com/jenkinsci/JenkinsPipelineUnit#mocking-jenkins-variables